src/tools/html-checker/main.rs

```diff
@@ -0,0 +1,65 @@
+use std::env;
+use std::path::Path;
+use std::process::{Command, Output};
+
+fn check_html_file(file: &Path) -> bool {
+    let mut to_mute = vec!["PROPRIETARY_ATTRIBUTE"];
+    if file.file_name().and_then(|f| f.to_str()) == Some("settings.html") {
+        // In `settings.html`, we have empty `<span>` for the toggles.
+        to_mute.push("TRIM_EMPTY_ELEMENT");
+    }
+    let to_mute_s = to_mute.join(",");
+    let mut command = Command::new("tidy");
+    command
+        .arg("-e")
+        .arg("-q")
+        .arg("--mute-id") // this option is useful in case we want to mute more warnings
+        .arg("yes")
+        .arg("--mute")
+        .arg(&to_mute_s)
+        .arg(file);
+
+    let Output { status, stderr, .. } = command.output().expect("failed to run tidy command");
+    if status.success() {
+        true
+    } else {
+        let stderr = String::from_utf8(stderr).expect("String::from_utf8 failed...");
+        if stderr.is_empty() {
+            true
+        } else {
+            eprintln!(
+                "=> Errors for `{}` (error code: {}) <=",
+                file.display(),
+                status.code().expect("failed to get exit code")
+            );
+            eprintln!("{}", stderr);
+            false
+        }
+    }
+}
+
+fn find_all_html_files(dir: &Path) -> usize {
+    let mut errors = 0;
+
+    for entry in dir.read_dir().expect("read_dir failed") {
+        let entry = entry.expect("Error on read_dir entry").path();
+        if entry.is_dir() {
+            errors += find_all_html_files(&entry);
```

Comment:
I would use `walkdir` instead of rewriting this from scratch, it's faster and already in our dependency tree.

<details><summary>cargo tree -i walkdir</summary>

```
> cargo tree -i walkdir
walkdir v2.3.1
├── cargo v0.54.0 (/home/joshua/src/rust/rustc2/src/tools/cargo)
│   └── rls v1.41.0 (/home/joshua/src/rust/rustc2/src/tools/rls)
├── cargo-util v0.1.0 (/home/joshua/src/rust/rustc2/src/tools/cargo/crates/cargo-util)
│   ├── cargo v0.54.0 (/home/joshua/src/rust/rustc2/src/tools/cargo) (*)
│   ├── cargo-test-support v0.1.0 (/home/joshua/src/rust/rustc2/src/tools/cargo/crates/cargo-test-support)
│   │   [dev-dependencies]
│   │   └── cargo v0.54.0 (/home/joshua/src/rust/rustc2/src/tools/cargo) (*)
│   └── rls v1.41.0 (/home/joshua/src/rust/rustc2/src/tools/rls)
├── compiletest v0.0.0 (/home/joshua/src/rust/rustc2/src/tools/compiletest)
├── ignore v0.4.16
│   ├── bootstrap v0.0.0 (/home/joshua/src/rust/rustc2/src/bootstrap)
│   ├── cargo v0.54.0 (/home/joshua/src/rust/rustc2/src/tools/cargo) (*)
│   └── rustfmt-nightly v1.4.37 (/home/joshua/src/rust/rustc2/src/tools/rustfmt)
│       └── rls v1.41.0 (/home/joshua/src/rust/rustc2/src/tools/rls)
├── installer v0.0.0 (/home/joshua/src/rust/rustc2/src/tools/rust-installer)
├── lint-docs v0.1.0 (/home/joshua/src/rust/rustc2/src/tools/lint-docs)
├── rls v1.41.0 (/home/joshua/src/rust/rustc2/src/tools/rls)
└── tidy v0.1.0 (/home/joshua/src/rust/rustc2/src/tools/tidy)
    └── unstable-book-gen v0.1.0 (/home/joshua/src/rust/rustc2/src/tools/unstable-book-gen)
[build-dependencies]
└── error_index_generator v0.0.0 (/home/joshua/src/rust/rustc2/src/tools/error_index_generator)
```

</details>

Reply:
Already scheduled. ;)